app/team.go

```diff
@@ -93,7 +91,17 @@ func (a *App) isTeamEmailAllowed(user *model.User, team *model.Team) bool {
 		return true
 	}
 	email := strings.ToLower(user.Email)
-	return a.isTeamEmailAddressAllowed(email, team.AllowedDomains)
+	allowedDomains := a.getAllowedDomains(user, team)
+	return a.isEmailAddressAllowed(email, allowedDomains)
+}
+
+func (a *App) getAllowedDomains(user *model.User, team *model.Team) []string {
+	if user.IsGuest() {
+		return []string{*a.Config().GuestAccountsSettings.RestrictCreationToDomains}
+	} else {
+		// First check per team allowedDomains, then app wide restrictions
+		return []string{team.AllowedDomains, *a.Config().TeamSettings.RestrictCreationToDomains}
+	}
```

Comment:
wouldn't it look cleaner as:
```go
result := []string{*a.Config().GuestAccountsSettings.RestrictCreationToDomains}
if !user.IsGuest() {
		// First check per team allowedDomains, then app wide restrictions
		result = append([]string{team.AllowedDomains}, result...)
	}
return result
```

Reply:
I tend to avoid re-assigning variable state but maybe this way is better or idiomatic for go. What do you think @agnivade ?

Reply:
Idiomatic Go is to just outdent the else condition:

```go
if user.IsGuest() {
	return []string{*a.Config().GuestAccountsSettings.RestrictCreationToDomains}
} 
// First check per team allowedDomains, then app wide restrictions
return []string{team.AllowedDomains, *a.Config().TeamSettings.RestrictCreationToDomains}
```